include/detail/aglShaderHolder.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include <container/seadPtrArray.h>
+#include <heap/seadDisposer.h>
+#include <hostio/seadHostIOReflexible.h>
+#include <prim/seadSafeString.h>
+#include <resource/seadArchiveRes.h>
+#include "agl/common/aglShaderProgramArchive.h"
+
+namespace agl {
+
+class ShaderProgram;
+
+namespace detail {
+
+class ShaderHolder {
+    SEAD_SINGLETON_DISPOSER(ShaderHolder);
+
+public:
+    ShaderHolder();
+    virtual ~ShaderHolder();
+
+    void initialize(sead::ArchiveRes* archiveRes, sead::Heap* heap);
+    sead::FormatFixedSafeString<1024> genMessage(sead::hostio::Context* context);
+    void listenPropertyEvent(const sead::hostio::PropertyEvent* property_event);
+
+private:
+    sead::FixedPtrArray<ShaderProgram, 206> mShaderPrograms;
+    ShaderProgramArchive mShaderProgramArchives[6];
```

Comment:
Nintendo seems to never use bare arrays in agl. They either use sead::SafeArray or sead::UnsafeArray. (Even for local variables, it seems.)

This should be a sead::UnsafeArray due to the iterator being used, but no bound checking.